cmd/loadtest/submitter/submitter.go

```diff
@@ -87,7 +88,20 @@ func (s *Submitter) StartSubmission(ctx context.Context, limiter *rate.Limiter,
 			close(s.Stopped)
 		}()
 
+		if native {
+			level.Debug(logger).Log("msg", "using native asset")
+		} else {
+			level.Debug(logger).Log("msg", "using non-native asset")
+		}
+
+		destIndex := 0
+
 		for {
+			destIndex++
+			if destIndex == len(s.destinationAddresses) {
+				destIndex = 0
+			}
```

Comment:
```golang
for {
    for i := 0; i < len(s.xxx); i++ {
        ...
    }
}
```

Reply:
Nesting a `for` messes with the error handling that indicates the cancellation from the duration timer.  Its simplest to leave this logic as-is.